src/lib.rs

```diff
@@ -31,37 +31,36 @@ pub fn ffprobe(path: impl AsRef<std::path::Path>) -> Result<FfProbe, FfProbeErro
         ])
         .arg(path)
         .output()
-        .map_err(|err| FfProbeError::new(format!("Could not execute ffprobe: {}", err)))?;
+        .map_err(FfProbeError::Io)?;
 
     if !out.status.success() {
-        let stderr = String::from_utf8_lossy(&out.stderr);
-
-        return Err(FfProbeError::new(format!(
-            "ffprobe exited with status code {}: {}",
-            out.status, stderr
-        )));
+        return Err(FfProbeError::Status(out));
     }
 
-    serde_json::from_slice::<FfProbe>(&out.stdout)
-        .map_err(|err| FfProbeError::new(format!("Could not deserialize ffprobe output: {}", err)))
+    serde_json::from_slice::<FfProbe>(&out.stdout).map_err(FfProbeError::Deserialize)
 }
 
 #[derive(Debug)]
-pub struct FfProbeError {
-    message: String,
-}
-
-impl FfProbeError {
-    fn new(message: impl Into<String>) -> Self {
-        Self {
-            message: message.into(),
-        }
-    }
+pub enum FfProbeError {
```

Comment:
Let's add `non_exhaustive`. I don't see what additional errors we could have, but it's better to have an option for non-breaking minor releases.

```suggestion
#[non_exhaustive]
pub enum FfProbeError {
```